benchmark/SerializerBenchmark/SerializerBenchmark.csproj

```diff
@@ -40,6 +40,12 @@
       <Private>true</Private>
       <SpecificVersion>false</SpecificVersion>
     </Reference>
+    <Reference Include="OdinSerializer">
+      <HintPath>OdinSerializer.dll</HintPath>
+      <Aliases>odin</Aliases>
+      <Private>true</Private>
+      <SpecificVersion>false</SpecificVersion>
+    </Reference>
```

Comment:
Can you use the `PackageReference` pattern above instead? We only use `Reference` for the old version of MessagePack to get around NuGet limitations.
That will probably solve your missing DLL problem.

Reply:
Odin isn't available on NuGet, unfortunately.

Reply:
Well, I guess you'll need to add a reference to the missing PInvoke assembly as well then, or use `None` items set to copy local to arrange the dlls to be deployed.

Reply:
How can I do that? I've never done this before.

Reply:
```xml
<PackageReference Include="System.Runtime.InteropServices.PInvoke" Version="4.0.0-rc2-24027" />
<Reference Include="OdinSerializer.dll">
  <Private>true</Private>
</Reference>
```

That said, I see `System.Runtime.InteropServices.PInvoke` is delisted on nuget, so I suspect OdinSerializer is misguided in depending on it.

Reply:
The fact remains that Odin is a major serialization library (at least as far as Unity is concerned), so I see no choice but to just deal with it. Let me try your suggestion and get back to you, thank you.

Reply:
I can't see either of the problems you report. It works when I check out your branch. I'm making a couple of touch-up changes and may merge this PR. Was there anything else you wanted to do to it?

Reply:
Ok, I see the PInvoke exception thrown at the tail end. Apparently although there is no assembly reference to it, the library may require it by reflection.

Reply:
Wow. It has a static constructor that reflects over all assemblies in the appdomain.

```cs
static FormatterLocator()
	{
		LOCK = new object();
		FormatterInstances = new Dictionary<Type, IFormatter>(FastTypeComparer.Instance);
		TypeFormatterMap = new DoubleLookupDictionary<Type, ISerializationPolicy, IFormatter>(FastTypeComparer.Instance, ReferenceEqualityComparer<ISerializationPolicy>.Default);
		FormatterLocators = new List<FormatterLocatorInfo>();
		FormatterInfos = new List<FormatterInfo>();
		Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
```

During reflection it probably stumbles over an assembly whose dependencies are not fully satisfied.

Reflecting over all loaded assemblies is such an anti-pattern, particularly from a static constructor, that I really don't care how fast it may be... I'm not using it, and we're probably not going to take this PR unless they fix this design first.
But based on how they don't push to nuget and they check in all kinds of SDKs into their repo rather than consume nuget packages, my guess is they aren't too keen on following common guidelines.

Reply:
> I can't see either of the problems you report. It works when I check out your branch.

In that case, it's entirely possible that something is weird with my own .NET installation. Any idea what it might be in case it becomes an unrelated problem for me later?

> I'm making a couple of touch-up changes and may merge this PR.

Thank you, can't wait!

> Was there anything else you wanted to do to it?

Nope, I'm good to go. I'd just appreciate you updating the numbers in the README to account for Odin.

Reply:
Whoops, you replied literally as I was. Jinx!

> Wow. It has a static constructor that reflects over all assemblies in the appdomain.
> 
> ```cs
> static FormatterLocator()
> 	{
> 		LOCK = new object();
> 		FormatterInstances = new Dictionary<Type, IFormatter>(FastTypeComparer.Instance);
> 		TypeFormatterMap = new DoubleLookupDictionary<Type, ISerializationPolicy, IFormatter>(FastTypeComparer.Instance, ReferenceEqualityComparer<ISerializationPolicy>.Default);
> 		FormatterLocators = new List<FormatterLocatorInfo>();
> 		FormatterInfos = new List<FormatterInfo>();
> 		Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
> ```
> 
> During reflection it probably stumbles over an assembly whose dependencies are not fully satisfied.
> 
> Reflecting over all loaded assemblies is such an anti-pattern, particularly from a static constructor, that I really don't care how fast it may be... I'm not using it, and we're probably not going to take this PR unless they fix this design first.
> But based on how they don't push to nuget and they check in all kinds of SDKs into their repo rather than consume nuget packages, my guess is they aren't too keen on following common guidelines.

I know, and I agree that it's a mess, but it's still an interesting comparison to make just because Odin is so popular among Unity developers (and that MessagePack specifically includes Unity support).

Reply:
Sure, it's interesting. And if you pursue this and find the answer, posting it to your original github issue may be useful to others. But I think putting pressure on OdinSerializer to fix their anti-patterns may be even more beneficial to the community.

Reply:
Can you at least help me run the benchmarks (including Odin) -- or run it on yours and post the results, since it seems to work for you -- since ultimately I'm trying to use this information to decide whether or not to switch to MessagePack?

Reply:
No. It's broken because of the anti-pattern it implements. I don't have time to fix someone else's library, sorry.

Reply:
I thought you said it worked when you checked it out? Or did I misinterpret something?

Reply:
I did say that, and it did when I commented out all the other serializers. But once it was combined with others so you could fairly compare them, it quit working. Probably because one of these other serializers are the party with an unresolved PInvoke reference.
So if you want to pursue this, you could try commenting out the other serializers (or subsets of them) till you get a combination that works.

Reply:
Fair enough. Thank you anyway, I appreciate your time.